docs/DeveloperGuide.md

```diff
@@ -1,34 +1,630 @@
+
+Given below are sequence diagrams describing the general Application flow and how the different objects
+interact with each other.
+
+![Initialization](img/InitializationSequenceDiagram.png)\
+_Figure 2: Initialization_
+
+The initialization sequence diagram above shows the systematic flow of object creation and record
+retrieval from storage file for creation of the `RecordList` object.
+
+![Main program flow](img/CommandLooperSequenceDiagram.png)\
+_Figure 3: Main Application Loop & Exit Sequence_
+
+> 💡 The lifeline all objects should end at the destroy marker (X) but due to a limitation of PlantUML,
+> the lifeline reaches the end of diagram.
+
+This sequence diagram follows suit after initialization in _Figure 2_.\
+This shows the main flow until the `exit` command is input by the user.
+
+The following sections below will provide more details of each component.
+
+### 3.2 UI Component
+
+![UIComponentDiagram](img/UIComponentDiagram.png)\
+*Figure 4: **Ui** Class Diagram*
+
+The Ui Component consists of a `Ui` class which handles all user input and system output.
+The Ui is only dependent on the `Duke` class and does not interact directly with other classes,
+ensuring a high level of cohesiveness, low level of coupling and separation of roles.
+
+The `Ui` component actively listens for:
+* the execution of commands to print the result of a `command`
+
+### 3.3 Parser Component
+![ParserHandlerClassDiagram](img/ParserHandlerClassDiagram.png) <br>
+_Figure 5: **ParserHandler** Class Diagram_
+
+#### Description
+The Parser component consist of 1 class called `ParserHandler`.
+The role of `ParserHandler` can be interpreted as a "manager" or "middle man" to parse
+the user input from the console into an `ArrayList<String>` format.
+
+#### Design
+In the main program, `ParserHandler` is instantiated through the constructor `new ParserHandler()`,
+finalizing the startOptionArray, endOptionArray, and middleOptionArray. Whenever the program needs
+to parse a user input, the ParserHandler calls the method `getParseInput` and returns an `ArrayList<String>`.
+
+1. `getParseInput` will trim the leading white space before calling `startExtraction`.
+2. `startExtraction` will check if the trimmed input starts with option. If yes, extract the option and 
+   remove the option from the trimmed input before calling `extractSubsequencePart`.
+3. `extractSubsequencePart` will check for the next option index and extract whatever is in between the start of the 
+   trimmed input to the start of the next option index. This is the argument tagged after the option. 
+   * Any leading or trailing white space of the argument field will be removed.
+   * If no argument is provided, the argument would be stored an empty string.
+
+   Afterward, the checking mechanism will loop until no valid next options are left in the input before calling
+   `extractFinalPart`.
+   * Valid next option format: `' <option> '` with 1 leading and trailing whitespace. e.g. `' -e '`
```

Comment:
Not your fault, but it seems like the github pages theme breaks code formatting here. Maybe you could force the code block into a new line using `\`.